index.js

```diff
@@ -1,35 +1,65 @@
-/*!
- * forwarded
- * Copyright(c) 2014 Douglas Christopher Wilson
- * MIT Licensed
- */
-
-/**
- * Module exports.
- */
+'use strict'
 
-module.exports = forwarded
+var Processor = require('./lib/processor')
+var schemas = require('./lib/schemas')
 
 /**
  * Get all addresses in the request, using the `X-Forwarded-For` header.
  *
- * @param {Object} req
+ * @param {http.IncomingMessage} req
  * @api public
  */
 
-function forwarded(req) {
+module.exports = function forwarded (req, options) {
+  options = options || {}
+
+  var opts = {
+    // default to only common + standard
+    // array order matters here
+    schemas: options.schemas || [
```

Comment:
I would argue that typically in libraries such as this, it should be the standard only (i.e. `RFC 7239`).

however due to the long history of `X-Forwarded-*` and the relatively new age of `RFC 7239` it would make sense to consider both a *standard*.

in that light, both are the defacto default here, other scenarios would be:

- `RFC 7239` only: would essentially break every dependant's assumptions
- `X-Forwarded-*` only: would not be true to the standard and might even be problematic in the future.

this is precisely why the logic at [index.js#L52](https://github.com/jshttp/forwarded/blob/3e90938bf9fd143a995171e239dcd27d172c9af9/index.js#L52) overwrites properties in the sorted order of this schemas array, existing implementations of `X-Forwarded-*` will still operate identical to the past, and when `Forwarded` header is present, it counts as the authoritative source of information *as it is the standard*.